lib/class-sp-admin.php

```diff
@@ -172,7 +172,25 @@ public function settings_page() {
 					<input type="hidden" name="action" value="sp_settings" />
 					<?php wp_nonce_field( 'sp_settings', 'sp_settings_nonce' ); ?>
 					<p>
-						<input type="text" name="sp_host" value="<?php echo esc_url( SP_Config()->get_setting( 'host' ) ); ?>" style="width:100%;max-width:500px" />
+						<label for="sp_host"><?php esc_html_e( 'Host:', 'searchpress' ); ?></label>
+						<input type="text" name="sp_host" id="sp_host" value="<?php echo esc_url( SP_Config()->get_setting( 'host' ) ); ?>" style="width:100%;max-width:500px" />
+					</p>
+					<p>
+						<label for="sp_index"><?php esc_html_e( 'Index (optional):', 'searchpress' ); ?></label>
+						<input type="text" name="sp_index" id="sp_index" value="<?php echo esc_attr( SP_Config()->get_setting( 'index' ) ); ?>" style="width:100%;max-width:500px" />
+					</p>
+					<p>
+						<label for="sp_username"><?php esc_html_e( 'Username (optional):', 'searchpress' ); ?></label>
+						<input type="text" name="sp_username" id="sp_username" value="<?php echo esc_attr( SP_Config()->get_setting( 'username' ) ); ?>" style="width:100%;max-width:500px" />
+					</p>
+					<p>
+						<label for="sp_password"><?php esc_html_e( 'Password (optional):', 'searchpress' ); ?></label>
+						<input type="password" name="sp_password" id="sp_password" style="width:100%;max-width:500px" />
+						<br /><?php esc_html_e( 'Stored password is encoded and will not load back into this form.', 'searchpress' ); ?>
```

Comment:
it's encoded, but in a way that can be decoded, so we can still work with it. @kevinfodness has some good art for password updates in WP forms.